pwiz_tools/Skyline/Model/DocSettings/PeptideSettings.cs

```diff
@@ -2100,18 +2100,16 @@ private LibraryIonMobilityInfo GetLibraryDriftTimesForFilePath(LibKey[] targetIo
         public bool HasAnyLibraryIonMobilities(IEnumerable<LibKey> targetIons)
         {
             var targets = targetIons?.ToArray();
-            foreach (var lib in _libraries.Where(l => l != null))
+            return _libraries.Any(lib => lib != null && HasIonMobilities(lib, targets));
+        }
+
+        public static bool HasIonMobilities(Library lib, LibKey[] targetIons)
+        {
+            for (var i = 0; lib.TryGetIonMobilityInfos(targetIons, i, out var ionMobilities); i++) // Returns false when i> internal list length

```

Comment:
Wouldn't it be more efficient to use the variant of TryGetIonMobilityInfos that does not take a FileIndex?